### DISCUSSION TITLE
Mirth Docker Upgrade with Postgres DB backend

### DISCUSSION THREAD

**Discussion by rjcdoc** [nextgenhealthcare/connect#6247] [Q&A] [answered]
We are on Mirth Connect 3.12 and are planning an upgrade to 4.5.0.  right now we have a yaml file pointing to the 3.12 image.  Would upgrading to 4.5 Docker mean we would change docker yaml to point to 4.5 image and also update the extension jars and plugins with the versions appropriate for 4.5 Mirth?  Are there no upgrade script that need to be run against the database?

**Comment by rjkroll** [comment]


**Reply by rjcdoc** [reply]
But there is no install "per-say" when we invoke the new docker compose command?  It just runs the new docker container version with the new bins.  So, if there are database schema changes, how would those get applied?  Are you saying mirth checks the database schema every time it starts and then does want it needs to do to make sure the db schema is correct for the version it is running?

**Reply by jonbartels** [reply]
> Are you saying mirth checks the database schema every time it starts and then does want it needs to do to make sure the db schema is correct for the version it is running?

Exactly this. You can see it in the source code at https://github.com/nextgenhealthcare/connect/tree/development/server/dbconf .

**Reply by rjcdoc** [reply, marked as answer]
So at a high level for docker upgrade steps.

1. Download compatible Custom_JARS_Mirth.zip & mc_plugins_custom.zip for the version we are going to, 4.5.0
2. Update our stack.yml file to point to the new docker image for version 4.5.0
For Example: 
```
version: "3.12.0"
services:
  mc:
    image: nextgenhealthcare/connect:3.12.0

```
to
```
version: "4.5.0"
services:
  mc:
    image: nextgenhealthcare/connect:4.5.0

```
4. Then update the stack.yml to point to the new zip files for the version 4.5.0
```
 - CUSTOM_JARS_DOWNLOAD=<Path_to_4.5.0_Custom_JARS_Mirth.zip>
 - EXTENSIONS_DOWNLOAD=<Path_to_4.5.0_mc_plugins_custom.zip>

```
5.  Run the docker compose command and Mirth will update the Postgres DB Schema is required.
6. Test channels and scripts and make sure it is all run as expected.
7. Upgrade done.

**Reply by jonbartels** [reply]
@rjcdoc One important detail

> Run the docker compose command and Mirth will update the Postgres DB Schema is required.

If you're not sure the upgrade will go smoothly, make sure you do this against a COPY of your existing DB. Eg dont YOLO this into PROD without verfiying a clean update and having backups to revert to!

**Comment by rjcdoc** [comment]
The upgrade for the most part went fine.  We are troubleshooting one problem.  It seems a couple of the extensions are not loading properly.  SSL Manager and InterOpSuite which has the XCDA Sender in it.

A few of our channels depend on these extensions.  I don't see any errors in the logs about these failing to load.  But they are not showing up in the Mirth Administrator GUI.

Are there other places to look for error about extension loading?

**Reply by jonbartels** [reply]
You paid for these extensions. If they're not loading the best resource is official support from NextGen Success Community.

Check `appdata/extension.properties` are the extensions disabled there?

How are you loading/installing the extensions initially? Copying ZIPs to `custom-extensions` in your container?

**Reply by rjcdoc** [reply]
After looking into the startup script, entrpoint.sh,  I saw the script is looking to unzip the plugins file and looking for zip files in the root directory of the zip file.  If it doesn't find any files there it just silently moves on.  So there is no indication there is an error or problem loading the plugins.  The plugins just aren't loaded.

After I fixed the zip file, everything is working.
